mbuild/compound.py

```diff
@@ -871,10 +873,23 @@ def remove_bond(self, particle_pair):
             The pair of Particles to remove the bond between
 
         """
+        from mbuild.port import Port
         if self.root.bond_graph is None or not self.root.bond_graph.has_edge(*particle_pair):
             warn("Bond between {} and {} doesn't exist!".format(*particle_pair))
             return
         self.root.bond_graph.remove_edge(*particle_pair)
+        bond_vector = particle_pair[0].pos - particle_pair[1].pos
+        if np.allclose(bond_vector, np.zeros(3)):
+            warn("Particles {} and {} overlap! Ports will not be added."
+                 "".format(*particle_pair))
+            return
+        distance = np.linalg.norm(bond_vector)
+        particle_pair[0].parent.add(Port(anchor=particle_pair[0],
+                                         orientation=-deepcopy(bond_vector),
```

Comment:
I'll go back and check. In one of my test cases `bond_vector` variable was getting changed after creating the first `Port`, but it would probably be best to identify and fix that issue rather than using `deepcopy` here as a band-aid.

Reply:
Any luck hunting this down?